rcl/src/rcl/subscription.c

```diff
@@ -186,13 +186,19 @@ rcl_subscription_get_rmw_handle(const rcl_subscription_t * subscription)
 bool
 rcl_subscription_is_valid(const rcl_subscription_t * subscription)
 {
+  const rcl_subscription_options_t * options = rcl_subscription_get_options(subscription);
```

Comment:
Isn't this an infinite loop? `rcl_subscription_get_options` calls this function.